src/app/api/services/dot-interceptor/dot-interceptor.service.ts

```diff
@@ -0,0 +1,87 @@
+import { DotMessageService } from './../dot-messages-service';
+import { Injectable } from '@angular/core';
+import { RequestOptionsArgs, Response } from '@angular/http';
+
+import { Observable } from 'rxjs/Observable';
+
+import { ResponseView, CoreWebService, LoginService } from 'dotcms-js/dotcms-js';
+
+import { DotDialogService } from '../dot-dialog';
+import { DotRouterService } from '../dot-router-service';
+
+@Injectable()
+export class DotInterceptor {
+    constructor(
+        private coreWebService: CoreWebService,
+        private dotDialogService: DotDialogService,
+        private dotMessageService: DotMessageService,
+        private dotRouterService: DotRouterService,
+        private loginService: LoginService
+    ) {
+        this.dotMessageService.getMessages([
+            'dot.common.http.error.403.header',
+            'dot.common.http.error.403.message',
+            'dot.common.http.error.500.header',
+            'dot.common.http.error.500.message'
+        ]).subscribe();
+    }
+
+    /**
+     * Do request, show errors in the ui and do redirects
+     *
+     * @param {RequestOptionsArgs} options
+     * @returns {Observable<ResponseView>}
+     * @memberof DotInterceptor
+     */
+    request(options: RequestOptionsArgs): Observable<ResponseView> {
+        return this.coreWebService
+            .requestView(options)
+            .catch((err: ResponseView) => this.handleErrors(err))
+            .filter((res: ResponseView) => res !== null);
+    }
+
+    private handleErrors(err: ResponseView): Observable<ResponseView> {
+        this.callErrorHandler(err.response.status);
+        return Observable.of(null);
```

Comment:
I think is a better idea if we just throw a exception, I dont know how this exactly work in ts but I supposed that it worked